tools/analyzer_plugin/lib/src/util/prop_forwarding/forwarding_config.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:analyzer/dart/element/element.dart';
+
+/// A representation of an over_react consumer's configuration of which props classes to
+/// include or exclude when forwarding props.
+abstract class PropForwardingConfig {
```

Comment:
I kept going back and forth between combining this class into `ForwardedProps`, but it seemed cleaner to keep it separate 
a) from an organizational/conceptual perspective and 
b) because information we need to populate `ForwardedProps.propsClassBeingForwarded` isn't present everywhere we'd parse one of these configs, and would have to be passed through potentially multiple layers of functions